src/check/runner/Runner.ts

```diff
@@ -154,45 +154,48 @@ function check<Ts>(rawProperty: IRawProperty<Ts>, params?: Parameters<Ts>): unkn
  * Run the property, throw in case of failure
  *
  * It can be called directly from describe/it blocks of Mocha.
- * It does not return anything in case of success.
+ * No meaningful results are produced in case of success.
  *
- * WARNING: Has to be awaited
+ * WARNING: Returns a promise to be awaited if the property is asynchronous
  *
- * @param property - Asynchronous property to be checked
+ * @param property - Synchronous or asynchronous property to be checked
  * @param params - Optional parameters to customize the execution
  *
  * @remarks Since 0.0.7
  * @public
  */
-function assert<Ts>(property: IAsyncProperty<Ts>, params?: Parameters<Ts>): Promise<void>;
+function assert<Ts>(
+  property: IRawProperty<Ts> & Exclude<IRawProperty<Ts>, IAsyncProperty<Ts> | IProperty<Ts>>,
+  params?: Parameters<Ts>
+): Promise<void> | void;
```

Comment:
I'm not sure to understand the need behind this move of the overload higher in the ordering. It seems that with the current setup (the one before your PR) the overloads gets properly shown in IDEs like VSCode.

Here is what I get for `asyncProperty`:
![image](https://user-images.githubusercontent.com/5300235/155281263-92e5dd73-3dc3-4e95-be82-6b932bfe81cc.png)

And for `property`:
![image](https://user-images.githubusercontent.com/5300235/155281647-abbbbcc5-dff5-44e3-bce8-e2f8fb24b958.png)

Your problem mostly be related to things being written in the current documentation and not with the ordering of those overloads.

Additionally as it worked well up to now, and to limit risks of regressions I'd not add complexity (Exclude... were not needed in the past, they make typings harder to read for non-ts users) on the typings for not needed reasons.